nimbleclient/v1/restclient.py

```diff
@@ -291,15 +325,21 @@ def put(self, endpoint, **payload):
             logging.exception(error)
             raise ConnectionError(f"Error communicating with {self.hostname}")
 
-    def post(self, endpoint, **payload):
+    def post(self, endpoint, **params):
         """Wrapper for POST requests"""
 
+        # Translate metadata dict to REST key-value pairs.
+        # Ex: {'k1': 'k1', 'k2': 'v2'} ==> [{'key': 'k1', 'value': 'v1'}, {'key': 'k2', 'value': 'v2'}]
+        if 'metadata' in params:
+            params['metadata'] = [{'key': key, 'value': params['metadata'][key]} for key in params['metadata']]
+
         try:
             while 1:
+                logging.debug("Query Params: %s", json.dumps(params, indent=4))
```

Comment:
The payload can contain sensitive data that we shouldn't log. I would remove this statement.